### PR TITLE
use prototypes

### PR DESCRIPTION
**Pull request by yoshuawuyts**
So umm, yeah everything's different. Why? Because the prior version was kinda big and hard to deal with. Is this better? - I think so, although we've opted for a super low level API. Will it be nice to use? It works for me, I wonder how it'll work for others. Anywayyyyyy

## ch-ch-ch-changes
- implements prototype based API
- from `4kb` down to `1kb`
- less chance of bugs, now hyper focused on allowing 3rd party libs to work with vdom style rendering
- removes most of the other lifecycle methods because they can easily be added when needed through libs
- should function as a good base for library authors, for app level stuff we should probably write some wrappers (not included in this repo tho)
- includes example of wrapping `leaflet.js`; was kind easy so heyyyyy it seems to work C:

### CODE REVIEW

File: example/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  function updateTitle (evt) {
+    emit('update-title', evt.target.value)
+  }
+
+  function toPhiladelphia () {
+    emit('set-coords', [39.9526, -75.1652])
+  }
+
+  function toSeattle () {
+    emit('set-coords', [47.6062, -122.3321])
```
Comment (toddself) [inline example/index.js]: 👍

**Comment by yoshuawuyts** [comment]
Screenshot of the example - should probs put it on netlify once this PR is merged

screen shot 2017-04-01 at 00 47 33

**Comment by yoshuawuyts** [comment]
@toddself hmm, yeah not sure we should; kinda feel it would make the example way more complicated than it should be. Think by far the `placeholder` stuff was the most interesting; I just created https://github.com/yoshuawuyts/on-idle so that behavior can easily be added to new Nanocomponent instances. The library we were previously using wouldn't work too well with the prototype based version of Nanocomponent.

File: README.md [hunk 2]
```diff
@@ -2,221 +2,80 @@
...
-}, 1000)
+// Implementer API
+var Nanocomponent = require('nanocomponent')
+var html = require('bel')
+
+function MyButton () {
+  if (!(this instanceof MyButton)) return new MyButton()
+  this._color = null
+  Nanocomponent.call(this)
+}
```
Comment (bcomnes) [inline README.md]: Nice, dropping the return from the constructor makes the API more clear I think.

**Comment by yoshuawuyts** [comment]
Think a fair share of people have looked at this now; nothing major has come up - merging! :D

**Comment by yoshuawuyts** [comment]
`v3.0.0`

**Pull request merged**